src/Builder.php

```diff
@@ -372,12 +372,24 @@ protected function getTotalCount($results)
     {
         $engine = $this->engine();
 
+        $totalCount = $engine->getTotalCount($results);
+
         if (is_null($this->queryCallback)) {
-            return $engine->getTotalCount($results);
+            return $totalCount;
+        }
+
+        $ids = $engine->mapIds($results)->all();
+
+        if (count($ids) < $totalCount) {
+            $ids = $engine->keys(tap(clone $this, function ($builder) use ($totalCount) {
+                $builder->take(
+                    is_null($this->limit) ? $totalCount : min($this->limit, $totalCount)
```

Comment:
setting the limit will prevent Scout Driver/Engine from setting a default limit. e.g: Meilisearch has 20 records limit by default.